src/helpers/filesystem/create_file.ts

```
@@ -0,0 +1,58 @@
+  } catch (error) {
+    console.error("Error creating file:", error);
+    // Optionally, display an error message to the user
+  }
+}
+
+export function sanitizeFilename(filename: string) {
+  // Replace spaces with underscores
+  let sanitized = filename.replace(/\s+/g, "_");
+
+  // Remove or replace non-ASCII characters
+  sanitized = sanitized.replace(/[^\x00-\x7F]/g, "");
```

Biome

[error] Unexpected control character(s) in regular expression: \x00

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)